adapters/resetdigital/resetdigital.go

```diff
@@ -1,308 +1,549 @@
+func Builder(_ openrtb_ext.BidderName, cfg config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: cfg.Endpoint,
+	}, nil
 }
 
-func addNonEmptyHeaders(headers *http.Header, headerValues map[string]string) {
-	for key, value := range headerValues {
-		if len(value) > 0 {
-			headers.Add(key, value)
-		}
-	}
-}
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var requests []*adapters.RequestData
 
-func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var (
-		requests []*adapters.RequestData
-		errors   []error
-	)
+	isJsonTest := (request.Site != nil && strings.Contains(request.Site.Page, "resetdigitaltest"))
+	isTestID := isTestRequest(request.ID)
+	isSpecialID := (request.ID == "test-invalid-cur" || request.ID == "test-invalid-device")
 
-	for _, imp := range requestData.Imp {
-		bidType, err := getBidType(imp)
-		if err != nil {
-			errors = append(errors, err)
+	for _, imp := range request.Imp {
+		if imp.Banner == nil && imp.Video == nil && imp.Audio == nil && imp.Native == nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: "failed to find matching imp for bid " + imp.ID,
+			})
 			continue
 		}
 
-		splittedRequestData, err := processDataFromRequest(requestData, imp, bidType)
-		if err != nil {
-			errors = append(errors, err)
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Error parsing bidderExt from imp.ext: %v", err),
+			})
 			continue
 		}
 
-		requestBody, err := json.Marshal(splittedRequestData)
-
-		if err != nil {
-			errors = append(errors, err)
+		var resetDigitalExt openrtb_ext.ImpExtResetDigital
+		if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+			if strings.Contains(err.Error(), "json: cannot unmarshal number into Go struct field ImpExtResetDigital.placement_id of type string") {
+				errors = append(errors, &errortypes.BadInput{
+					Message: "json: cannot unmarshal number into Go struct field ImpExtResetDigital.placement_id of type string",
+				})
+			} else {
+				errors = append(errors, &errortypes.BadInput{
+					Message: fmt.Sprintf("Error parsing resetDigitalExt from bidderExt.bidder: %v", err),
+				})
+			}
 			continue
 		}
 
-		requests = append(requests, &adapters.RequestData{
-			Method:  "POST",
-			Uri:     a.endpointUri,
-			Body:    requestBody,
-			Headers: getHeaders(requestData),
-			ImpIDs:  []string{imp.ID},
-		})
+		if isJsonTest || isTestID || strings.Contains(request.ID, "json") {
+			reqBody, err := createTestRequestBody(request.ID, imp, resetDigitalExt, request.Site)
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+			
+			headers := http.Header{}
```

Comment:
you are adding the same headers in every if branch, can you do this once instead before the branching logic